File: snippets/java/Snippet-002.java
import java.math.BigDecimal;
import java.util.List;

class Product {
    private ProductType type;
    private BigDecimal price;

    public Product(ProductType type, BigDecimal price) {
        this.type = type;
        this.price = price;
    }

    public int getType() { return type.id; }

    public BigDecimal getPrice() {
        return price;
    }

    public boolean isFood() {
        return type == ProductType.Food;
    }
}

class Products {
    public static long countFoodProducts(List<Product> products) {
        return products.stream()
                .filter(Product::isFood)
                .count();
    }
}

enum ProductType {
    Other(0),
    Food(19),
    ;

    public final int id;

    ProductType(int id) {
        this.id = id;
    }
}

File: snippets/kotlin/Snippet-002.kt
data class Product(
    val type: Int,
    val price: Float
) {
    val foodType = 19

    fun isFood(): Boolean = this.type == foodType
}


fun getNumberOfProductThatAreAboutFood(list: List<Product>): Int {
    return list
        .filter { it.isFood() }
        .size
}
